arch/arm/boot/dts/ste-ux500-samsung-golden.dts

```diff
@@ -24,6 +24,32 @@
 		stdout-path = &serial2;
 	};
 
+	i2c-gpio-0 {
+		compatible = "i2c-gpio";
+		sda-gpios = <&gpio2 14 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		scl-gpios = <&gpio2 13 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&i2c_gpio_0_default>;
+
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		touchkey@20 {
+			compatible = "cypress,midas-touchkey";
+			reg = <0x20>;
+			vdd-supply = <&ab8500_ldo_aux6_reg>;
+			vcc-supply = <&ab8500_ldo_aux4_reg>;
```

Comment:
I suggest swapping these two regulators. The downstream kernel and schematic I have does not make clear which one of them is which, but the `vdd-supply` is used for LED brightness control in the driver:

https://github.com/novathor-mainline/linux/blob/231702365892778b7f27f9b8d572cce36b94140e/drivers/input/keyboard/tm2-touchkey.c#L86-L95

And note how it changes the voltage to 2.5V / 3.3V to turn the LED on/off. At the moment this causes errors every time you change the brightness:

```
[  141.629001] touch_key_2.2v: voltage operation not allowed
[  148.248833] touch_key_2.2v: voltage operation not allowed
[  152.262619] touch_key_2.2v: voltage operation not allowed
```

Therefore I think it is more appropriate to use the `key_led_3.3v` regulator (`ab8500_ldo_aux4_reg`) as `vdd-supply`.

Now the error still appears when you change the brightness to `0` (but not anymore for `1`):

```
[   50.498276] key_led_3.3v: voltage operation not allowed
[   68.321245] key_led_3.3v: voltage operation not allowed
```

This is because the regulators in `samsung-golden` have a fixed voltage, and should not be changed to another voltage. Fortunately, looking at the code above there is already a `fixed_regulator` option in the driver that prevents changing the regulator voltage.

Therefore I suggest that you:
1. Swap the two regulators as suggested above.
2. Define a new `tc360_touchkey_variant` that looks like `midas_touchkey_variant`,  
but with `.fixed_regulator = true`
3. Use it with a new `coreriver,tc360-touchkey` compatible

Another advantage of the `fixed_regulator` option is that it properly synchronizes the initial state of the touch key backlight. `brightness` is set to `1` after boot, but the backlight is actually off. With `fixed_regulator`, it gets enabled during bootup:

https://github.com/novathor-mainline/linux/blob/231702365892778b7f27f9b8d572cce36b94140e/drivers/input/keyboard/tm2-touchkey.c#L279-L280

Please split this into two commits, one for the input driver change, and one for the dts change.

If you need any help, please feel free to ask here or on Matrix/IRC. I tested this already but this is your patch and I don't want to spoil the fun ;)